src/TeachMeSkills/Homework/Homework3and4.java

```diff
@@ -0,0 +1,247 @@
+        }
+
+        // Одномерные массивы. Задание 6
+        System.out.print("Введите размер массива: ");
+        size = 0;
+        while (!sc.hasNextInt()){
+            System.out.println("Введите число!");
+            sc.next();
+        }
+        size = sc.nextInt();
+        array = new int[size];
+        for (int i = 0; i < array.length; i++) {
+            array[i] = (int) (Math.random() * 10);
+        }
+        System.out.println("Массив: " + Arrays.toString(array));
+        for (int i = 0; i < array.length; i++){
+            if (i % 2 != 0) {
+                array[i] = 0;
+            }
+        }
+        System.out.println("Новый массив: " + Arrays.toString(array));
+
+        // Многомерные массивы. Задание 1
+        String[][] arrayString = new String[8][8];
+        for (int i = 0; i < arrayString.length; i++)
+        {
+            for (int j = 0; j < arrayString[0].length; j++)
+            {
+                if ((i % 2 != 0 && j % 2 != 0) || (i % 2 == 0 && j % 2 == 0)) {
+                    arrayString[i][j] = "W";
+                } else {
+                    arrayString[i][j] = "B";
+                }
+            }
+            System.out.println(" " + Arrays.toString(arrayString[i]));
+        }
+
+        // Многомерные массивы. Задание 2
+        int n = 0;
+        int m = 0;
+        System.out.print("Введите высоту массива: ");
+        while (!sc.hasNextInt()){
+            System.out.println("Введите число!");
+            sc.next();
+        }
+        n = sc.nextInt();
+        System.out.print("Введите ширину массива: ");
+        m = sc.nextInt();
+        int[][] arrayMulti = new int[n][m];
+        System.out.println("Полученный массив:");
+        int sum = 0;
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < m; j++) {
+                arrayMulti[i][j] = (int) (Math.random() * 10);
+                sum+=arrayMulti[i][j];
+            }
+            System.out.println(" " + Arrays.toString(arrayMulti[i]));
+        }
+        System.out.println("Сумма всех элементов массива равна: " + sum);
+
+        // Многомерные массивы. Задание 3
+        System.out.print("Введите высоту массива: ");
+        while (!sc.hasNextInt()){
+            System.out.println("Введите число!");
+            sc.next();
+        }
+        int i = 0;
+        int j = 0;
+        n = sc.nextInt();
+        System.out.print("Введите ширину массива: ");
+        m = sc.nextInt();
+        int[][] arrayMulti1 = new int[n][m];
+        System.out.println("Полученный массив:");
+        for (i = 0; i < n; i++) {
+            for (j = 0; j < m; j++) {
+                arrayMulti1[i][j] = (int) (Math.random() * 10);
+            }
+            System.out.println(" " + Arrays.toString(arrayMulti1[i]));
+        }
+        System.out.print("Главная диагональ:");
+        for (i = 0; i < (Math.min(arrayMulti1[0].length, arrayMulti1.length)); i++) {
```

Comment:
It's better to move `Math.min(arrayMulti1[0].length, arrayMulti1.length)` to a variable